src/modules/adhs/src/lib/Adhs.c

```diff
@@ -0,0 +1,356 @@
+                    {
+                        currentMatch = fileContent + matchGroups[2].rm_so;
+                        currentMatchSizeBytes = matchGroups[2].rm_eo - matchGroups[2].rm_so;
+                        
+                        for (unsigned int i = 0; i < g_permissionConfigMapCount; i++)
+                        {
+                            if ((currentMatchSizeBytes == strlen(g_permissionConfigMapKeys[i])) && 
+                                (0 == strncmp(currentMatch, g_permissionConfigMapKeys[i], currentMatchSizeBytes)))
+                            {
+                                value = g_permissionConfigMapValues[i];
+                                break;
+                            }
+                        }
+                    }
+
+                    if (NULL == value) 
+                    {
+                        if (IsFullLoggingEnabled())
+                        {
+                            OsConfigLogError(AdhsGetLog(), "MmiGet failed to find valid TOML property '%s'", g_permissionConfigName);
+                        }
+                        status = EINVAL;
+                    }
+                }
+                else 
+                {
+                    if (IsFullLoggingEnabled())
+                    {
+                        OsConfigLogError(AdhsGetLog(), "MmiGet failed to find TOML property '%s'", g_permissionConfigName);
+                    }
+                    status = EINVAL;
+                }
+
+                regfree(&permissionRegex);
+            } 
+            else 
+            {
+                OsConfigLogError(AdhsGetLog(), "MmiGet failed to compile regular expression '%s'", g_permissionConfigPattern);
+                status = EINVAL;
+            }
+            
+            FREE_MEMORY(fileContent);
+        }
+        else 
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(AdhsGetLog(), "MmiGet failed to read TOML file '%s'", g_adhsConfigFile);
+            }
+            status = EINVAL;
+        }
+
+        // Reset status and payload if TOML file could not be parsed or property not found, as it may yet have to be configured.
+        if (MMI_OK != status)
+        {
+            value = g_permissionConfigMapValues[0];
+            status = MMI_OK;
+        }
+
+        *payloadSizeBytes = strlen(value);
+
+        if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+        {
+            OsConfigLogError(AdhsGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported value will be truncated", componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+            *payloadSizeBytes = g_maxPayloadSizeBytes;
+        }
+
+        *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+        if (NULL != *payload)
+        {
+            memcpy(*payload, value, *payloadSizeBytes);

```

Comment:
>memcpy(*payload, value, *payloadSizeBytes); [](http://example.com/codeflow?start=12&length=43)

add a memset to all 0 before just in case

Reply:
Yes, makes sense, done!